engine/src/test/scala/cromwell/engine/WorkflowDescriptorSpec.scala

```diff
@@ -2,89 +2,58 @@ package cromwell.engine
-  val configWithCallCachingOff = defaultConfig + ("call-caching.enabled" -> "false")
-  val backend = new CromwellTestkitSpec.TestWorkflowManagerSystem()
-
-  it should "honor configuration and workflow options for call-caching" in {
-    val configs = Seq(defaultConfig, configWithCallCachingOn, configWithCallCachingOff)
-    val options = Seq(None, Some(true), Some(false))
-
-    val allCombinations = (for {
-      config <- configs
-      writeOption <- options
-      readOption <- options
-    } yield (config, writeOption, readOption)).toSet
-
-    // writeCache is ON when config is ON and write-to-cache is None or true
-    val writeCacheOnCombinations = (for {
-      config <- configs if config == configWithCallCachingOn
-      writeOption <- options if writeOption.isEmpty || writeOption.get
-      readOption <- options
-    } yield (config, writeOption, readOption)).toSet
-
-    // readCache is ON when config is ON and read_from_cache is None or true
-    val readCacheOnCombinations = (for {
-      config <- configs if config == configWithCallCachingOn
-      writeOption <- options
-      readOption <- options if readOption.isEmpty || readOption.get
-    } yield (config, writeOption, readOption)).toSet
-
-    def makeOptions(writeOpt: Option[Boolean], readOpt: Option[Boolean]) = {
-      val writeValue = writeOpt map { v => s""""write_to_cache": $v""" }
-      val readValue = readOpt map { v => s""""read_from_cache": $v""" }
-      val workflowOptions = Seq(writeValue, readValue).flatten.mkString(",\n")
-
-      s"{$workflowOptions}"
-    }
+  implicit val awaitTimeout = CromwellTestkitSpec.timeoutDuration
+  implicit val actorSystem: ActorSystem
 
-    def makeWorkflowDescriptor(config: Map[String, String], write: Option[Boolean], read: Option[Boolean]) = {
-      val sources = WorkflowSourceFiles(dummyWdl, "{}", makeOptions(write, read))
-      WorkflowDescriptor(WorkflowId.randomId(), sources, ConfigFactory.parseMap(config.asJava))
-    }
+  val workflowDescriptorMaterializerActor: ActorRef
 
-    writeCacheOnCombinations foreach {
-      case (config, writeToCache, readFromCache) => makeWorkflowDescriptor(config, writeToCache, readFromCache).writeToCache shouldBe true
-    }
+  def materializeWorkflowDescriptorFromSources(id: WorkflowId = WorkflowId.randomId(),
+                                               workflowSources: WorkflowSourceFiles,
+                                               conf: Config = ConfigFactory.load): WorkflowDescriptor = {
+    import scala.concurrent.ExecutionContext.Implicits.global
 
-    readCacheOnCombinations foreach {
-      case (config, writeToCache, readFromCache) => makeWorkflowDescriptor(config, writeToCache, readFromCache).readFromCache shouldBe true
+    val wfDescFut = PromiseActor.askNoTimeout(MaterializeWorkflowDescriptorActor.MaterializeWorkflow(id, workflowSources, conf),
+      workflowDescriptorMaterializerActor).mapTo[MaterializationResult]
```

Comment:
Discussed and no change needed. @gauravs90 pointed out that the `Await` will timeout just below.